src/EFCore/Query/ExpressionVisitors/Internal/ParameterExtractingExpressionVisitor.cs

```diff
@@ -294,6 +294,22 @@ protected override Expression VisitMember(MemberExpression memberExpression)
             return ExtractParameters(queryable.Expression);
         }
 
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        protected override Expression VisitConditional(ConditionalExpression conditionalExpression)
+        {
+            var returnValue = (TryOptimize(conditionalExpression.Test) as ConstantExpression)?.Value as bool?;
+
+            if (returnValue == null)
+            {
+                return base.VisitConditional(conditionalExpression);
+            }
+
+            return (bool)returnValue ? conditionalExpression.IfTrue : conditionalExpression.IfFalse;
```

Comment:
Here you haven't visited IfTrue/IfFalse part of the expression.
```suggestion
            return Visit((bool)returnValue ? conditionalExpression.IfTrue : conditionalExpression.IfFalse);
```